pkg/cmd/deletecmd/delete_application.go

```diff
@@ -160,11 +179,12 @@ func (o *DeleteApplicationOptions) deleteProwApplication(repoService jenkinsv1.S
 		NS:           ns,
 		IgnoreBranch: true,
 	}
-	names, err := prowOptions.GetReleaseJobs()
+	jobs, err := prowOptions.GetReleaseJobs()
```

Comment:
So maybe 'GetReleaseJobs' is really the best we have in the case of _applications_. Feels wrong though. IMO there should be a clear separation between applications/projects and environments. At some stage, we need to try unravelling this instead of monkey patching. 

If we go for filtering, we might want to consider whether it makes sense to push the filtering into `GetReleaseJobs` instead of just applying only in _delete_application_. This sounds like something which could be useful at other places as well. Just thinking out loud atm. @ccojocar WDYT?

Reply:
Another disadvantage here is that this is tying us t Prow which is probably not really future proof (thinking Lighthouse here).

I am wondering whether there is some sort of label we could base a selection on which differs between a project/application created via jx [import|create] and say the env pipelines.

Or whether it is possible to first retrieve application namespaces (staging|production|...) and then find applications in this namespace. In this model the dev environment would not count as application namespace.

Reply:
Also, how does this compare with `jx get application`? Does that have the same problem and if so, do we need to align between them?

Reply:
@hferentschik good point. We seem to be getting a deployments list on `jx get application` which does return only the applications and not the environments.

https://github.com/jenkins-x/jx/blob/a2908f49f3b03b1374fc3e04ab57a93b1bee963c/pkg/cmd/get/get_applications.go#L281

What I wasn't able to figure out is how to get the repository owner out of the deployment. Do you know if there's a way?

Also, as you mention if we were to follow this approach then we'd only need to request the applications from a given environment to solve #5671

Reply:
See https://github.com/jenkins-x/jx/pull/6069/files#r345255299

Reply:
> What I wasn't able to figure out is how to get the repository owner out of the deployment. Do you know if there's a way?

Not sure I follow. What do you mean?

Reply:
@hferentschik right now the generated output with the delete command is:

```
[ ]  mauromorales/jx-qs-test-delete-apps
```

however from `GetDeployments` I only seem to be able to get the name of the application`jx-qs-test-delete-apps` and I'm not aware how to relate it to the repo url/owner